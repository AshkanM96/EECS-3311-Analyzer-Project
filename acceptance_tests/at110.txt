open_set_enumeration
open_set_enumeration
integer_constant (2)
integer_constant (3)
add
integer_constant (1)
integer_constant (1)
integer_constant (4)
close_set_enumeration
open_set_enumeration
integer_constant (2)
negative
negative
integer_constant (3)
multiply
integer_constant (2)
integer_constant (2)
close_set_enumeration
close_set_enumeration
evaluate
type_check
reset
open_set_enumeration
open_set_enumeration
integer_constant (2)
integer_constant (3)
add
integer_constant (1)
integer_constant (1)
integer_constant (4)
close_set_enumeration
open_set_enumeration
integer_constant (4)
negative
negative
integer_constant (2)
multiply
integer_constant (-3)
integer_constant (-1)
close_set_enumeration
close_set_enumeration
evaluate
type_check
reset
