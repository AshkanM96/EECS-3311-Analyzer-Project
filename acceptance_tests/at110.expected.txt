  Expression currently specified: ?
  Report: Expression is initialized.
->open_set_enumeration
  Expression currently specified: {?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{?}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, ?}}
  Report: OK.
->integer_constant(3)
  Expression currently specified: {{2, 3, ?}}
  Report: OK.
->add
  Expression currently specified: {{2, 3, (? + nil)}}
  Report: OK.
->integer_constant(1)
  Expression currently specified: {{2, 3, (1 + ?)}}
  Report: OK.
->integer_constant(1)
  Expression currently specified: {{2, 3, (1 + 1), ?}}
  Report: OK.
->integer_constant(4)
  Expression currently specified: {{2, 3, (1 + 1), 4, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {?}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, ?}}
  Report: OK.
->negative
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- ?)}}
  Report: OK.
->negative
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- ?))}}
  Report: OK.
->integer_constant(3)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), ?}}
  Report: OK.
->multiply
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (? * nil)}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * ?)}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2), ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2)}, ?}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2)}}
  Report: OK.
->evaluate
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2)}}
  Report: {{2, 3, 4}}
->type_check
  Expression currently specified: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2)}}
  Report: {{2, 3, (1 + 1), 4}, {2, (- (- 3)), (2 * 2)}} is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
->open_set_enumeration
  Expression currently specified: {?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{?}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, ?}}
  Report: OK.
->integer_constant(3)
  Expression currently specified: {{2, 3, ?}}
  Report: OK.
->add
  Expression currently specified: {{2, 3, (? + nil)}}
  Report: OK.
->integer_constant(1)
  Expression currently specified: {{2, 3, (1 + ?)}}
  Report: OK.
->integer_constant(1)
  Expression currently specified: {{2, 3, (1 + 1), ?}}
  Report: OK.
->integer_constant(4)
  Expression currently specified: {{2, 3, (1 + 1), 4, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {?}}
  Report: OK.
->integer_constant(4)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, ?}}
  Report: OK.
->negative
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- ?)}}
  Report: OK.
->negative
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- ?))}}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), ?}}
  Report: OK.
->multiply
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (? * nil)}}
  Report: OK.
->integer_constant(-3)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * ?)}}
  Report: OK.
->integer_constant(-1)
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1), ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1)}, ?}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1)}}
  Report: OK.
->evaluate
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1)}}
  Report: {{2, 3, 4}}
->type_check
  Expression currently specified: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1)}}
  Report: {{2, 3, (1 + 1), 4}, {4, (- (- 2)), (-3 * -1)}} is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
